### ISSUE TITLE
entropy checkpointing can be optimized when using "entropy_from_logits_with_chunking" function

### ISSUE THREAD

**Issue by hailuo97** [volcengine/verl#2448]
https://github.com/volcengine/verl/blob/ab11fff33dcaa2409e388ce2f19aff440a5b703f/verl/workers/actor/dp_actor.py#L201
The torch.utils.checkpoint.checkpoint function is applied to the entire "entropy_from_logits_with_chunking" function. During the backward, it first recalculates the forward of the entire "for" loop, which causes the accumulation of GPU memory. This process can be modified as follows, move the torch.utils.checkpoint.checkpoint in "for" loop

def entropy_from_logits(logits: torch.Tensor):
    """Calculate entropy from logits."""
    pd = torch.nn.functional.softmax(logits, dim=-1)
    entropy = torch.logsumexp(logits, dim=-1) - torch.sum(pd * logits, dim=-1)
    return entropy

def entropy_from_logits_with_chunking(logits: torch.Tensor, chunk_size: int = 2048):
    """Memory-efficient entropy calculation with chunking."""
    entropy = torch.zeros(logits.shape[0], device=logits.device)
    for i in range(0, logits.shape[0], chunk_size):
        logits_chunk = logits[i : i + chunk_size].float()
        entropy_chunk = torch.utils.checkpoint.checkpoint(entropy_from_logits, logits_chunk)
        entropy[i : i + chunk_size] = entropy_chunk
    return entropy

**Comment by Kirrito-k423** [comment]
### 中文
定时清理检查结论：这个 issue 创建于 `2025-07-10T03:19:16Z`，截至本轮运行时间 `2026-07-10T01:03:01Z` 已经超过 6 个月；comments 数量为 0，并且在 timeline API `https://api.github.com/repos/verl-project/verl/issues/2448/timeline` 中未发现公开引用、PR/issue cross-reference、commit/PR mention 或其它公开关联互动。检查来源还包括 issue comments API `https://api.github.com/repos/verl-project/verl/issues/2448/comments`。

因此按社区 issue 定时清理标准关闭。若作者或社区成员仍能提供当前版本的复现、日志、PR 关联或新的需求背景，欢迎继续评论请求重开，或创建一个带有最新证据的新 issue。

### English
Scheduled cleanup check: this issue was created at `2025-07-10T03:19:16Z`, which is more than 6 months before this run time `2026-07-10T01:03:01Z`; it has 0 comments, and the timeline API `https://api.github.com/repos/verl-project/verl/issues/2448/timeline` did not show any public references, PR/issue cross-references, commit/PR mentions, or other public linked activity. The issue comments API `https://api.github.com/repos/verl-project/verl/issues/2448/comments` was also checked.

Closing this under the scheduled issue cleanup policy. If the author or community can still provide a current reproduction, logs, PR reference, or updated requirement context, please comment to request reopening or open a new issue with fresh evidence.

This response was generated by a scheduled Codex GPT5.5 issue processing task.
